pkg/kubectl/cmd/util/cmdtools.go

```diff
@@ -73,34 +81,40 @@ func defaultEnvShell() []string {
 	return []string{shell, flag}
 }
 
-func defaultEnvEditor(envs []string) ([]string, bool) {
-	var editor string
+func defaultEnv(cmdName string, envs []string) ([]string, bool) {
+	var cmd string
 	for _, env := range envs {
 		if len(env) > 0 {
-			editor = os.Getenv(env)
+			cmd = os.Getenv(env)
 		}
-		if len(editor) > 0 {
+		if len(cmd) > 0 {
 			break
 		}
 	}
-	if len(editor) == 0 {
-		editor = platformize(defaultEditor, windowsEditor)
+	if len(cmd) == 0 {
+		switch cmdName {
+		case "editor":
+			cmd = platformize(defaultEditor, windowsEditor)
+		case "diff":
+			cmd = checkGitDiffInstalled(gitDiff, defaultDiff)
```

Comment:
sorry, that naming was a little confusing, the defaultDiff means system default tool, not default using it, but I'll change the parameter order, or we could change name, like gitDiff, Diff ?